### DISCUSSION TITLE
Tour component

### DISCUSSION THREAD

**Discussion by Newbie012** [chakra-ui/chakra-ui#9249] [Ideas]
### Description

An interactive UI feature that guides users through key elements or functionalities of an application, often using tooltips, modals, or overlays; Zag's [tour state machine](https://zagjs.com/components/react/tour) provides a foundation for implementing such tours, though some functionality is still missing.

### Problem Statement/Justification

Implementing a custom tour could be a bit complicated to work alongside Chakra-UI when navigating through overlay components such as Drawer (either the overlay is closed, or the tour is inaccessible due to focused trap of the overlayed component)

### Proposed Solution or API

It seems like there's already a proposed solution in Zag, but it currently missing some functions such as `setSteps`. In addition, it doesn't behave well when opening and closing drawers/dialogs during the tour.

**Comment by segunadebayo** [comment]
This is planned. Will be added to Ark first then will land in Chakra

**Reply by Newbie012** [reply]
Thanks. Is it possible to take into consideration the ability to use the tour across overlay components? when I used Zag's tour, it messed up with the focus trap logic.

**Reply by 7alip** [reply]
@segunadebayo I noticed that the tour component is available in [Ark](https://ark-ui.com/docs/components/tour). Do you have any plans to include it in the roadmap?

**Reply by segunadebayo** [reply]
Yes, it's landing in Chakra pretty soon.

We have a PR opened for this

**Comment by segunadebayo** [comment]
Closing as stale. Please start a new discussion if this is still relevant on the latest version.
